File: POO - Periodo2/Atividade_Extra_POO_2/Questao 7/entities/ControleDeAudio.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ControleDeAudio = void 0;
var ControleDeAudio = /** @class */ (function () {
    function ControleDeAudio(volume) {
        this.minVolume = 0;
        this.maxVolume = 10;
        this.volume = volume;
    }
    ControleDeAudio.prototype.aumentarVolume = function () {
        if (this.volume < this.maxVolume) {
            this.volume += 1;
        }
    };
    ControleDeAudio.prototype.aumentarVolumeVariasVezes = function (qtdVezes) {
        if (this.volume < this.maxVolume) {
            this.volume += qtdVezes;
            if (this.volume > this.maxVolume) {
                this.volume = this.maxVolume;
            }
        }
    };
    ControleDeAudio.prototype.diminuirVolume = function () {
        if (this.volume > this.minVolume) {
            this.volume -= 1;
        }
    };
    ControleDeAudio.prototype.diminuirVolumeVariasVezes = function (qtdVezes) {
        if (this.volume > this.minVolume) {
            this.volume -= qtdVezes;
            if (this.volume < this.minVolume) {
                this.volume = this.minVolume;
            }
        }
    };
    ControleDeAudio.prototype.lerVolume = function () {
        return this.volume;
    };
    return ControleDeAudio;
}());
exports.ControleDeAudio = ControleDeAudio;

File: POO - Periodo2/Atividade_Extra_POO_2/Questao 5/src/entities/ControleDeAudio.java
package entities;

public class ControleDeAudio {
    private int volume;
    private final int minVolume = 0;
    private final int maxVolume = 10;

    public ControleDeAudio(int volume){
        this.volume = volume;
    }

    public void aumentarVolume(){
        if (this.volume < this.maxVolume){
            this.volume += 1;
        }
    }

    public void aumentarVolumeVariasVezes(int qtdVezes){
        if (this.volume < this.maxVolume){
            this.volume += qtdVezes;
            if (this.volume > this.maxVolume){
                this.volume = this.maxVolume;
            }
        }
    }

    public void diminuirVolume(){
        if (this.volume > this.minVolume){
            this.volume -= 1;
        }
    }

    public void diminuirVolumeVariasVezes(int qtdVezes){
        if (this.volume > this.minVolume){
            this.volume -= qtdVezes;
            if (this.volume < this.minVolume){
                this.volume = 0;
            }
        }
    }

    public int lerVolume(){
        return this.volume;
    }
}